### DISCUSSION TITLE
Configuration validation fails on self-hosted instance

### DISCUSSION THREAD

**Discussion by R3DST0RM** [renovatebot/renovate#17061] [Request Help] [answered]
I'm using Renovate as a Self-Hosted instance on my private GitLab server. The following configuration file was working until version `32.137.0`. Afterwards, the validation failed - there were no changes to the config.js file.

More information on my issue and the respective config files:

```js
module.exports = {
    endpoint: '<private endpoint>',
    platform: 'gitlab',
    enabledManagers: ['npm'],
    onboardingConfig: {
        extends: ['config-base']
    },
    repositories: [
        'foo/bar',
    ],
    platformAutomerge: true,
    packageRules: [
        {
            matchDepTypes: ['dependencies'],
            matchUpdateTypes: ['patch', 'pin'],
            automerge: true
        },
        {
            matchDepTypes: ['devDependencies'],
            matchUpdateTypes: ['minor', 'patch', 'pin'],
            automerge: true
        }
    ],
    hostRules: [
        {
            matchHost: '<private npm>',
            hostType: 'npm',
            username: 'user env',
            password: 'pass env'
        }
    ],
    commitMessagePrefix: 'Renovate',
    labels: ['renovate'],
    branchTopic: '{{{depNameSanitized}}}-{{{newVersion}}}',
    npmrc: 'npmrc env'
}
``` 
The gitlab-ci job is using the latest Docker image and therefore the configuration for the job is as follows:

```yml
image: renovate/renovate:latest

stages:
  - update

dependency-update:
  stage: update
  script:
    - renovate-config-validator config.js
    - renovate
```

After renovate version `32.137.0` I am experiencing the following error:

```
DEBUG: Using RE2 as regex engine
 WARN: config.js could not be parsed
       "err": {
         "code": "MODULE_NOT_FOUND",
         "requireStack": [
           "/usr/src/app/node_modules/renovate/dist/workers/global/config/parse/file.js",
           "/usr/src/app/node_modules/renovate/dist/config-validator.js",
           "/usr/src/app/dist/config-validator.js"
         ],
         "message": "Cannot find module 'config.js'\nRequire stack:\n- /usr/src/app/node_modules/renovate/dist/workers/global/config/parse/file.js\n- /usr/src/app/node_modules/renovate/dist/config-validator.js\n- /usr/src/app/dist/config-validator.js",
         "stack": "Error: Cannot find module 'config.js'\nRequire stack:\n- /usr/src/app/node_modules/renovate/dist/workers/global/config/parse/file.js\n- /usr/src/app/node_modules/renovate/dist/config-validator.js\n- /usr/src/app/dist/config-validator.js\n    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)\n    at Function.Module._load (internal/modules/cjs/loader.js:746:27)\n    at Module.require (internal/modules/cjs/loader.js:974:19)\n    at require (internal/modules/cjs/helpers.js:101:18)\n    at /usr/src/app/node_modules/renovate/lib/workers/global/config/parse/file.ts:22:25\n    at getParsedContent (/usr/src/app/node_modules/renovate/lib/workers/global/config/parse/file.ts:22:25)\n    at /usr/src/app/node_modules/renovate/lib/config-validator.ts:61:31"
       }
```

So I am wondering, what am I doing wrong? I could not find anything related to this issue.

**Comment by rarkins** [comment]
That does not appear to be running within the Renovate container. Otherwise why is there `C:\\` files?

**Reply by R3DST0RM** [reply]
It’s running inside the docker container. That’s the wrong stack trace, it was from a local reproduction repo.

```bash
DEBUG: Using RE2 as regex engine
 WARN: config.js could not be parsed
       "err": {
         "code": "MODULE_NOT_FOUND",
         "requireStack": [
           "/usr/src/app/node_modules/renovate/dist/workers/global/config/parse/file.js",
           "/usr/src/app/node_modules/renovate/dist/config-validator.js",
           "/usr/src/app/dist/config-validator.js"
         ],
         "message": "Cannot find module 'config.js'\nRequire stack:\n- /usr/src/app/node_modules/renovate/dist/workers/global/config/parse/file.js\n- /usr/src/app/node_modules/renovate/dist/config-validator.js\n- /usr/src/app/dist/config-validator.js",
         "stack": "Error: Cannot find module 'config.js'\nRequire stack:\n- /usr/src/app/node_modules/renovate/dist/workers/global/config/parse/file.js\n- /usr/src/app/node_modules/renovate/dist/config-validator.js\n- /usr/src/app/dist/config-validator.js\n    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)\n    at Function.Module._load (internal/modules/cjs/loader.js:746:27)\n    at Module.require (internal/modules/cjs/loader.js:974:19)\n    at require (internal/modules/cjs/helpers.js:101:18)\n    at /usr/src/app/node_modules/renovate/lib/workers/global/config/parse/file.ts:22:25\n    at getParsedContent (/usr/src/app/node_modules/renovate/lib/workers/global/config/parse/file.ts:22:25)\n    at /usr/src/app/node_modules/renovate/lib/config-validator.ts:61:31"
       }
```

**Comment by rarkins** [comment, marked as answer]
The validation didn't before support CLI parameters, e.g. config.js. Now the feature has been added.

I think if you remove config.js then it may just work, otherwise ensure that you provide the full absolute path to the file.

**Reply by R3DST0RM** [reply]
Removing config.js a parameter did the trick!

**Comment by nishasati6oct** [comment]
I am getting the below error for Self Hosted renovate in AKS cluster and self hosted GH app for renovate

Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "991b0053-b3f9-49fb-a8db-d022afdec357",
           "repository": "org/repo",
           "err": {
             "message": "platform-unknown-error",
             "stack": "Error: platform-unknown-error\n    at Proxy.initRepo (/usr/src/app/node_modules/renovate/lib/modules/platform/github/index.ts:522:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at getPlatformConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/apis.ts:98:26)\n    at initApis (/usr/src/app/node_modules/renovate/lib/workers/repository/init/apis.ts:110:12)\n    at initRepo (/usr/src/app/node_modules/renovate/lib/workers/repository/init/index.ts:56:12)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:64:14)\n    at attributes.repository (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:206:11)\n    at start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:191:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:19:22"
           },
           "msg": "Repository has unknown error"
         }
       ]
ERROR: renovate cli error (repository=org/repo)
       "err": {
         "cmd": "/bin/sh -c /usr/bin/node_default /usr/src/app/node_modules/renovate/dist/renovate.js",
         "stderr": "",
         "stdout": "",
         "options": {
           "encoding": "utf-8",
           "env": [
             "KUBERNETES_SERVICE_PORT_HTTPS",
             "RENOVATE_BINARY_SOURCE",
             "MEND_RNV_DATA_HANDLER_TYPE",
             "MEND_RNV_ENQUEUE_JOBS_ON_STARTUP",
             "MEND_RNV_CRON_JOB_SCHEDULER_ALL",
             "KUBERNETES_SERVICE_PORT",
             "USER_ID",
             "RENOVATE_X_EXEC_GPID_HANDLE",
             "LOG_LEVEL",
             "HOSTNAME",
             "MEND_RNV_WORKER_CLEANUP_DIRS",
             "PGPORT",
             "PGPASSWORD",
             "RENOVATE_MEND_RENOVATE_CE_PORT_80_TCP_ADDR",
             "MEND_RNV_WEBHOOK_SECRET",
             "MEND_RNV_POSTGRES_SSL_PEM_PATH",
             "ENV",
             "PWD",
             "MEND_RNV_LICENSE_KEY",
             "MEND_RNV_ACCEPT_TOS",
             "MEND_RNV_SERVER_API_SECRET",
             "MEND_RNV_RENOVATE_VERSION",
             "USER_NAME",
             "HOME",
             "USER_HOME",
             "LANG",
             "KUBERNETES_PORT_443_TCP",
             "RENOVATE_MEND_RENOVATE_CE_SERVICE_PORT",
             "MEND_RNV_GITHUB_APP_ID",
             "MEND_RNV_ENDPOINT",
             "PGUSER",
             "RENOVATE_MEND_RENOVATE_CE_SERVICE_PORT_HTTP",
             "MEND_RNV_GITHUB_APP_KEY",
             "PIP_DISABLE_PIP_VERSION_CHECK",
             "RENOVATE_MEND_RENOVATE_CE_PORT",
             "MEND_RNV_SQLITE_EXTENSIONS_PATHS",
             "SHLVL",
             "BASH_ENV",
             "RENOVATE_MEND_RENOVATE_CE_PORT_80_TCP",
             "KUBERNETES_PORT_443_TCP_PROTO",
             "PGDATABASE",
             "CONTAINERBASE",
             "MEND_RNV_ADMIN_API_ENABLED",
             "MEND_RNV_PLATFORM",
             "KUBERNETES_PORT_443_TCP_ADDR",
             "MEND_RNV_AUTODISCOVER_FILTER",
             "RELEASE_VERSION",
             "MEND_RNV_REPORTING_ENABLED",
             "RENOVATE_MEND_RENOVATE_CE_PORT_80_TCP_PORT",
             "PGHOST",
             "DOCKERFILE_NODE_DEFAULT_PATH",
             "KUBERNETES_SERVICE_HOST",
             "LC_ALL",
             "KUBERNETES_PORT",
             "KUBERNETES_PORT_443_TCP_PORT",
             "RENOVATE_MEND_RENOVATE_CE_SERVICE_HOST",
             "PATH",
             "CONTAINERBASE_ENV",
             "MEND_RNV_SYNC_MODE",
             "RENOVATE_MEND_RENOVATE_CE_PORT_80_TCP_PROTO",
             "DEBIAN_FRONTEND",
             "MEND_RNV_PROMETHEUS_METRICS_ENABLED",
             "GIT_SSH_COMMAND",
             "RENOVATE_AUTODISCOVER",
             "RENOVATE_CONFIG",
             "LOG_FILE"
           ],
           "maxBuffer": null,
           "stdio": "inherit",
           "timeout": 3600000
         },
         "exitCode": 1,
         "name": "ExecError",
         "message": "Command failed: /usr/bin/node_default /usr/src/app/node_modules/renovate/dist/renovate.js\n",
         "stack": "ExecError: Command failed: /usr/bin/node_default /usr/src/app/node_modules/renovate/dist/renovate.js\n\n    at ChildProcess.<anonymous> (/usr/src/app/node_modules/renovate/dist/util/exec/common.js:88:24)\n    at ChildProcess.emit (node:events:536:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at ChildProcess._handle.onexit (node:internal/child_process:293:12)"
       }
DEBUG: Renovate job finished (repository=org/repo)
       "status": "cli-error"
       
Here is my Helm chart values file
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: renovate
  namespace: renovate
spec:
  chart:
    spec:
      chart: mend-renovate-ce
      version: "=9.4.0"
      sourceRef:
        kind: HelmRepository
        name: mend-renovate-ce-ee
        namespace: flux-system
  install:
    remediation:
      retries: 3
  upgrade:
    remediation:
      retries: 3
  interval: 1m0s
  releaseName: renovate
  values:
    image:
      repository: ghcr.io/mend/renovate-ce
      tag: 9.4.0-full
      pullPolicy: IfNotPresent

    renovate:
      # Additional env vars
      extraEnvVars: 
        - name: MEND_RNV_POSTGRES_SSL_PEM_PATH
          value: /etc/config/log_level.conf

      # You must accept the Mend Terms of Service to use the image.
      # Please read https://www.mend.io/terms-of-service/
      # Set this value to 'y' to consent.
      mendRnvAcceptTos: 'y'

      # Which platform Mend Renovate will connect to.
      # valid values: "bitbucket-server", "github" or "gitlab"
      mendRnvPlatform: "github"

      # API endpoint, include trailing slash. Examples: https://api.github.com/, https://gitlab.com/api/v4/
      mendRnvEndpoint: https://api.github.com/

      mendRnvAutoDiscoverFilter: 'org/repo'

      # Optional: Provide a path to persist the SQLite database (eg. '/db/renovate-ce.sqlite', where 'db' is defined as a volume)
      # If you set cachePersistence.enabled to true, the default value for this setting will persist the SQLite database automatically
      mendRnvSqliteFilePath: ''

      mendRnvAdminApiEnabled: true

      # Existing secret with secret values with the following keys:
      #   mendRnvLicenseKey:
      #   mendRnvGitlabPat:
      #   mendRnvGithubAppId:
      #   mendRnvGithubAppKey:
      #   mendRnvWebhookSecret:
      #   mendRnvServerApiSecret:
      #   githubComToken:
      #   pipIndexUrl:
      existingSecret: renovate-admin-creds

      # Optional, defaults to '0 * * * *' (hourly)
      mendRnvCronJobSchedulerAll: '*/5 * * * *' #setting it in every 5 mins for testing. later we can set it daily once if needed

      # Optional, defaults to '0 0,4,8,12,16,20 * * *' (every 4 hours)
      mendRnvCronAppSync:

      # defaults to 'true'
      mendRnvSyncOnStartup:

      # defaults to 'bulk'
      # GitHub users only: can be set to 'batch'
      mendRnvSyncMode: 'batch'

      # Optional: Set to 'true' to enable Reporting APIs. Defaults to 'false'.
      mendRnvReportingEnabled: true

      mendRnvPrometheusMetricsEnabled: true

      # Optional. Accepted values: 'enabled', 'discovered', and 'disabled'. Defaults to 'discovered'.
      mendRnvEnqueueJobsOnStartup: disabled

      # Self-hosted renovate configuration file, will be mounted as a config map
      config: |
        module.exports = {
          // Enter self-hosted configuration options here.
          // https://docs.renovatebot.com/self-hosted-configuration/
        }
      # Set log level, defaults to 'info'. Allowed values: fatal, error, warn, info, debug, trace
      logLevel: debug

    postgresql:
      # if to use PostgreSQL, if set to 'false' SQLite will be used instead. We can enable it later to use.
      enabled: true
      host: idpbackstage.postgres.database.azure.com
      port: 5432
      database: renovate
      user: backstageadm

    ## Cache Persistence Parameters
    ## ref: https://kubernetes.io/docs/user-guide/persistent-volumes/
    ##
    cachePersistence:
      ## Enable persistence using Persistent Volume Claims
      enabled: true
      ## Persistent Volume storage class
      ## If defined, storageClassName: <storageClass>
      ## If set to "-", storageClassName: "", which disables dynamic provisioning
      ## If undefined (the default) or set to null, no storageClassName spec is set, choosing the default provisioner
      storageClass: "default"
      ## @param Persistent Volume access modes
      accessModes:
        - ReadWriteOnce
      ## Persistent Volume size
      size: 2Gi

    service:
      type: ClusterIP
      port: 80

    ingress:
      enabled: true
      ingressClassName: nginx
      hosts:
        - renovate.xxxx.com
      paths:
        - path: "/webhook"
          pathType: ImplementationSpecific
      tls: 
        - secretName: renovate.xxxx.com
          hosts: 
            - renovate.xxxx.com


    nodeSelector:
      agentpool: systemnp
    tolerations:
      - key: CriticalAddonsOnly
        operator: Exists

    podSecurityContext:
      fsGroup: 1000
    # This allows renovate to finish running for a repo and then gracefully exit
    terminationGracePeriodSeconds: 60

    livenessProbe:
      initialDelaySeconds: 10
      httpGet:
        path: /health
        port: http
      periodSeconds: 30
      timeoutSeconds: 1
      failureThreshold: 3
      successThreshold: 1

    readinessProbe:
      httpGet:
        path: /health
        port: http
      periodSeconds: 30
      timeoutSeconds: 1
      failureThreshold: 3
      successThreshold: 1

    extraVolumes:
      - name: pgsql-cert
        configMap:
          name: pgsql-cert
          items:
            - key: DigiCertGlobalRootCA.crt.pem
              path: log_level.conf
    
    extraVolumeMounts:
      - name: pgsql-cert
        mountPath: "/etc/config"

I am not getting the reason behind the cli error. Please provide some guidance.
